digitalocean/resource_digitalocean_spaces_bucket.go

```diff
@@ -81,6 +81,24 @@ func resourceDigitalOceanBucket() *schema.Resource {
 					},
 				},
 			},
+			// This is structured as a subobject in case Spaces supports more of s3.VersioningConfiguration
+			// than just enabling bucket versioning.
+			"versioning": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+					},
+				},
+			},
```

Comment:
```suggestion
			"versioning": {
				Type:     schema.TypeList,
				Optional: true,
				MaxItems: 1,
				Elem: &schema.Resource{
					Schema: map[string]*schema.Schema{
						"enabled": {
							Type:     schema.TypeBool,
							Optional: true,
							Default:  false,
						},
					},
				},
				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
					return old == "1" && new == "0"
				},
			},
```

In some testing, I think this should do what we want. By removing `Computed`, it will detect the need for an update when removing a block with `versioning.enabled = true` but the `DiffSuppressFunc` will prevent a diff like:

```diff
      - versioning {
          - enabled = false -> null
        }
```

The AWS provider has something similar (though not used with the `s3_bucket` resource for some reason): 

https://github.com/terraform-providers/terraform-provider-aws/blob/98b8b848ca94031b20c3e626c9d40484e3af80de/aws/diff_suppress_funcs.go#L36-L42

Reply:
That should work. I'll update the acceptance test to also test the case of having versioning enabled but going back to not specifying it.